src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,86 @@
+import { CSSProperties, KeyboardEvent } from "react";
+import { Close } from "@material-ui/icons";
+import styled from "styled-components";
+import _IconButton from "../IconButton";
+
+interface DialogProps {
+  open: boolean;
+  disableCloseButton?: boolean;
+  onClose?: () => void;
+  children?: React.ReactNode;
+  className?: string;
+  style?: CSSProperties;
+}
+
+export default function Dialog({
+  open,
+  disableCloseButton = false,
+  onClose = () => {},
+  children,
+  ...props
```

Comment:
IMO, a spread operator for props like; `...props` should be used when you are extending other types. e.g. 
```
interface SomeProps extends ButtonProps {
  label: string; 
  onPress?: () => void;
}

function someProps({label, onPress, ...buttonProps}: SomeProps): JSX.Element {
  return (
    <>
        ...
    </>
  )
}
``` 
You are simply burying property values with the spreading operator which will affect readability. So IMO, as long as you use properties that you defined yourself, you should explicitly call it when you can.

Reply:
there's a relevant eslint rule to refer: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md